lib/elixir/test/elixir/typespec_test.exs

```diff
@@ -120,14 +120,46 @@ defmodule TypespecTest do
       end
     end
 
-    test "undefined spec for function" do
+    test "type variable only used once (singleton type variable)" do
+      assert_raise CompileError, ~r"type variable x is only used once", fn ->
```

Comment:
Did this raise before or are we introducing a compile error that did not raise before?